### ISSUE TITLE
MinimumTextContrastGuideline fails when button/label contrast appears to be OK

### ISSUE THREAD

**Issue by HansMuller** [flutter/flutter#22044]
I've introduced some significant changes to ButtonTheme et al. in https://github.com/flutter/flutter/pull/22013.

With these changes, three RaisedButton contrast tests from examples/flutter_gallery/test/accessibility_test.dart fail. 

The buttons appear to be rendered correctly and with sufficient contrast. The tests computed foreground and background colors aren't what I'd expect, they should be background: 0xff0175c2 (see examples/flutter_gallery/lib/gallery/themes.dart primaryColor in buildLightTheme), foreground: white.

![show_bottom_sheet](https://user-images.githubusercontent.com/1377460/45771232-70e35e80-bbf9-11e8-9264-9057ba3b8e42.png)

```
// 'modal_bottom_sheet_demo'
The following TestFailure object was thrown running a test:
  Expected: Text contrast should follow WCAG guidelines
  Actual: <Instance of 'WidgetTester'>
   Which: SemanticsNode#786(Rect.fromLTRB(265.0, 304.0, 535.0, 352.0), actions: [tap], flags:
   [isButton, hasEnabledState, isEnabled], label: "SHOW BOTTOM SHEET", textDirection: ltr):
          Expected contrast ratio of at least 4.5 but found 2.65 for a font size of 14.0. The
          computed foreground color was: Color(0xff6200ee), The computed background color was:
          Color(0xff0d0020)
          See also: https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html

// 'persistent_bottom_sheet_demo'
The following TestFailure object was thrown running a test:
  Expected: Text contrast should follow WCAG guidelines
  Actual: <Instance of 'WidgetTester'>
   Which: SemanticsNode#893(Rect.fromLTRB(279.0, 168.0, 521.0, 216.0), actions: [tap], flags:
   [isButton, hasEnabledState, isEnabled], label: "SHOW A SNACKBAR", textDirection: ltr):
          Expected contrast ratio of at least 4.5 but found 2.65 for a font size of 14.0. The
          computed foreground color was: Color(0xff6200ee), The computed background color was:
          Color(0xff0d0020)
          See also: https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html

// 'snack_bar_demo'
The following TestFailure object was thrown running a test:
  Expected: Text contrast should follow WCAG guidelines
  Actual: <Instance of 'WidgetTester'>
   Which: SemanticsNode#786(Rect.fromLTRB(265.0, 304.0, 535.0, 352.0), actions: [tap], flags:
   [isButton, hasEnabledState, isEnabled], label: "SHOW BOTTOM SHEET", textDirection: ltr):
          Expected contrast ratio of at least 4.5 but found 2.65 for a font size of 14.0. The
          computed foreground color was: Color(0xff6200ee), The computed background color was:
          Color(0xff0d0020)
          See also: https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html
```

**Comment by HansMuller** [comment]
cc @jonahwilliams

**Comment by jonahwilliams** [comment]
Running the test on a device produced the following image, so it looks like the test logic is okay. Do the demos require some additional theming now?

<img width="350" src="https://user-images.githubusercontent.com/8975114/45850750-25f64380-bcec-11e8-9f96-124d8802d85a.png">

**Comment by HansMuller** [comment]
That's quite possible, will see if I can track down the problem.

**Comment by HansMuller** [comment]
In the '..contrast guidelines' tests,  the test demo's appearance didn't match the gallery because the test used the default "light" theme, rather than one of the Gallery's themes. So were testing the a11y of the default light theme (good) but not the gallery's actual default theme.

https://github.com/flutter/flutter/pull/22141 tests both the default light theme and the gallery's light theme. To close this issue we'll also need to test the corresponding dark themes.  That part of #22141 is disabled for now because the text contrast computation appears to (sometimes) get the wrong fg/bg colors.

**Comment by jonahwilliams** [comment]
I believe this is fixed by https://github.com/flutter/flutter/pull/31000
